plugins/pubIds/urn/URNPubIdPlugin.php

```diff
@@ -533,33 +536,45 @@ public function loadUrnFieldComponent($hookName, $args)
             return;
         }
 
-        $templateMgr->addJavaScript(
-            'urn-field-component',
-            Application::get()->getRequest()->getBaseUrl() . '/' . $this->getPluginPath() . '/js/FieldUrn.js',
-            [
-                'contexts' => 'backend',
-                'priority' => TemplateManager::STYLE_SEQUENCE_LAST,
-            ]
-        );
+        $context = $templateMgr->getTemplateVars('currentContext');
+        $suffixType = $this->getSetting($context->getId(), 'urnSuffix');
+        if ($suffixType === 'default' || $suffixType === 'pattern') {
+            $templateMgr->addJavaScript(
+                'field-pub-id-urn-component',
+                Application::get()->getRequest()->getBaseUrl() . '/' . $this->getPluginPath() . '/js/FieldPubIdUrn.js',
+                [
+                    'contexts' => 'backend',
+                    'priority' => TemplateManager::STYLE_SEQUENCE_LAST,
+                ]
+            );
+        } else {
+            $templateMgr->addJavaScript(
+                'field-text-urn-component',
+                Application::get()->getRequest()->getBaseUrl() . '/' . $this->getPluginPath() . '/js/FieldTextUrn.js',
+                [
+                    'contexts' => 'backend',
+                    'priority' => TemplateManager::STYLE_SEQUENCE_LAST,
+                ]
+            );
```

Comment:
These two JavaScript files should be combined into one, so there is only one request made.  Alternatively, you can pass an additional argument along with `contexts` and `priority`:

```
'inline' => true
```

That will print it directly into a `<script>` tag so no extra request is required.

Reply:
So should maybe all of them be combined into one file? :thinking:

Reply:
Hihihihi :-) I combined all 3 into one file.
I just haven't managed to use inline => true -- then I get errors:

Reply:
Ah, I have just realized that checkNumber.js is needed for other objects, like issue and galley -- they still use the old template... Thus, I will restore and use all 3 different js -- the two vue components will be loaded as needed...

Reply:
inline => true does not work -- I would need to put the script code instead of script path/url as the second argument for the function templateManager->addJavaScript().
Thus, leaving it as it here...

Reply:
Ok, no problem to leave it as-is. :+1: